### PR TITLE
Awesome Books (Phase 2)

### PR DESCRIPTION
**Pull request by katarighe**
In this phase, we have done the following tasks: 

1. **Used** JavaScript `classes` instead of `objects` and `arrays`. 
2. **Also used** `objects` and `arrays` but is included inside the classes.
3. **Created** class methods that adds and removes books. 
4. **After refactoring**, we **make sure** that everything is working normally as it should and that our application is free from errors and bugs. 
5. **Not using** any library or framework, just only plain JavaScript.
6. **Added** CSS styles to make the wireframe match to the sample layout. 

**Sample layout:**

image

**Our layout:**

image

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,63 +1,67 @@
-// Make sure if the local storage in empty than add an empty array
-if (localStorage.getItem('Added Books') == null) {
-  localStorage.setItem('Added Books', JSON.stringify([]));
-}
+class BookCollection {
+  // Make sure if the local storage in empty than add an empty array
+  static getStoredBooks() {
```
Comment (souad988) [inline index.js:3]: Good job was done so far 👍🏼  , I want just to mention that Static methods are often utility functions, such as functions to create or clone instances. Public static fields are useful when you want a field to exist only once per class, not on every class instance you create , here you need to have list of books as property of the class and it should be initialized in the constructor of the class in this case your BookCollection class can be reusable you can create different instances to manages different categories of books for example 
and onload you create a new instance of your BookCollection class

`constructor() {
    this.books = [];
  }
`

File: index.js [hunk 2]
```diff
@@ -1,63 +1,67 @@
...
+  static addNewBook(bookTitle, bookAuthor) {
+    const storedBooks = BookCollection.getStoredBooks();
+    const newBook = {
+      title: bookTitle,
+      author: bookAuthor,
+    };
+    storedBooks.push(newBook);
+    BookCollection.updateStoredBooks(storedBooks);
+    BookCollection.displayBooks(storedBooks);
+  }
```
Comment (souad988) [inline index.js]: Same as the previews comment this method is meant to be adding to the new instance of books , and by adding the constructor you can now access the books by using the key word this 

``this.books....``

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @katarighe , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

242722065-db428b0c-8dfc-49c4-a156-42f227e32868


## Highlights
Refactor code to use classes
Fair collaboration
No linter errors
Document code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 1 above)
Comment (katarighe) [reply index.js:3]: @souad988 I'm confused about this part. Can you explain step by step on this?

**Review by uwadonat — APPROVED** [review]
## Hi @Team,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- You are using the correct workflow ✔️
- Well-organized and clean project ✔️
- Professional readme file  ✔️
approv

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**